### PR TITLE
Join button

### PR DESCRIPTION
**Pull request by jsbsmp**
### Checklist
- [x] I've run `mvn test` from the root directory to see all new and existing tests pass
- [x] I've followed code style and run and ensured the code style is valid
- [x] I've created new tests if necessary.

### Motivation and Context
https://github.com/ctco-dev/carpooling-2018/issues/23

### Description
Added functionality to the button "Join" (from Main page) for reducing trip places number in active trips table by click.
If there are no available seats, the button "Join" is unavailable.
Created tests for new methods.

### CODE REVIEW

File: src/main/java/lv/ctco/javaschool/app/control/TripStore.java [hunk 1]
```diff
@@ -30,4 +31,21 @@
...
     }
 
+    public Optional<Trip> findTripById(Long id) {
+        return em.createQuery("select t from Trip t where t.id = :id", Trip.class)
+                .setParameter("id", id)
+                .getResultStream()
+                .findFirst();
+    }
+
+    public void setTrip(int places, String id) {
```
Comment (olegvm) [inline src/main/java/lv/ctco/javaschool/app/control/TripStore.java]: The method name should match its purpose, e.g. `setTripPlaces()`.

File: src/main/webapp/javascript/main.js [hunk 2]
```diff
@@ -28,4 +27,27 @@ function logout() {
...
 function goMyProfile() {
     location.href = "/profile.jsp";
+}
+function join(button) {
+    var data = {};
+    var places = $(button).closest('tr').find('.table_places').text() - 1;
+    var tripId = $(button).closest('tr').find('.table_id').text();
+    var id = $(button).closest('td').parent().index();
+    var x = document.getElementById("trips").rows[id + 1].cells;
+    x[4].innerHTML = places;
```
Comment (olegvm) [inline src/main/webapp/javascript/main.js]: The variable name `x` does not explain its purpose. Please use more meaningful names.
E.g. `places` and `tripId` are OK names, while `id` does not make it clear whose id it is, and `x` is an unknown.

File: src/main/webapp/javascript/main.js [hunk 3]
```diff
@@ -28,4 +27,27 @@ function logout() {
 }
 function goMyProfile() {
     location.href = "/profile.jsp";
+}
+function join(button) {
+    var data = {};
+    var places = $(button).closest('tr').find('.table_places').text() - 1;
+    var tripId = $(button).closest('tr').find('.table_id').text();
+    var id = $(button).closest('td').parent().index();
```
Comment (olegvm) [inline src/main/webapp/javascript/main.js]: This code will easily break if something changes in DOM structure. Can you pass these variables as parameters to the `join()` method?`

**Comment by jsbsmp** [comment]
- [x] renamed setTrip() method of the TripStore class to setTripPlaces()
- [x] renamed variable x to more meaningful name "rowCells"
- [x] converted variables tripId, rowId and places to join() method parameters

File: src/main/webapp/main.jsp [hunk 4]
```diff
@@ -35,13 +36,16 @@
...
             <td>{{from}}-{{to}}</td>
             <td>{{driverInfo}}</td>
             <td>{{driverPhone}}</td>
-            <td>{{places}}</td>
+            <td class="table_places" id="places">{{places}}</td>
             <td>{{event}}</td>
             <td>
-                <button>Join</button>
+                <button id="join-button" type="button"
+                        onclick="join(this, $(this).closest('tr').find('.table_id').text(), $(this).closest('td').parent().index(), $(this).closest('tr').find('.table_places').text())">Join
```
[the remark this answers is not in the document]
Comment (olegvm) [reply src/main/webapp/main.jsp]: As per discussion, plain IDs won't work here because the markup is repeated for each row acquired from DB.

**Pull request merged**